File: backend/schemas/knowledge.py
"""知识复盘 + 错题归因 schema（PRD v1.4 板块二，P0 合规整改后）。

与旧版差异（对齐 docs/openapi.yaml v1.5）：
- ErrorParseRequest：不再接收 question_text/student_answer/correct_answer 原文，
  改为引用已入库错题（errorId）；原文只本地检索、永不出域。
- KnowledgeSummaryCreate：知识复盘请求改为结构化入参（学科/周期），
  不再接收前端自由拼装的 error_summary/mastery_changes/state_context 文本。
  v2.2 起改为 periodStart/periodEnd/subject 并异步落库 summaries 表。
"""
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class KnowledgeSummaryCreate(BaseModel):
    """知识复盘请求（POST /api/v1/knowledge-summary）。"""

    model_config = ConfigDict(populate_by_name=True)

    subject: str = Field(..., alias="subject", description="学科，如 math / 数学")
    period: str = Field(..., alias="period", description="复盘周期，如「本周」")


class KnowledgeSummaryResponse(BaseModel):
    """知识复盘响应。"""

    summary: str = Field(..., description="LLM 生成的知识复盘文案")


class ErrorParseRequest(BaseModel):
    """错题归因请求（合规形态）。"""

    model_config = ConfigDict(populate_by_name=True)

    error_id: str = Field(..., alias="errorId", description="已入库错题 ID（err_ 前缀），原文不出域")


class ErrorParseResponse(BaseModel):
    """错题归因响应。"""

    parse: str = Field(..., description="LLM 生成的智能解析")

File: frontend/src/services/knowledge.ts
/**
 * 板块二 · 知识库相关接口封装（演示用）。
 *
 * 真实后端在 v2.1/v2.2 阶段实现，本文件先按契约声明入参/出参。
 * 现阶段后端未上线，所有调用都会因网络层失败抛出 ApiError，
 * 由 UI 层捕获并 toast 提示。
 */

import { apiPost } from './http';

/* ============ 复盘生成 ============ */

export interface KnowledgeSummaryPayload {
  subject: string;
  period: string;
  error_summary: string;
  mastery_changes: string;
  state_context: string;
}

export interface KnowledgeSummaryResponse {
  summary: string;
  /** 是否来自 LLM；false 即规则模板降级 */
  from_llm?: boolean;
  /** 生成时间 ISO 串 */
  generated_at?: string;
}

/** POST /api/knowledge-summary —— 学科阶段复盘 */
export function generateKnowledgeSummary(
  payload: KnowledgeSummaryPayload,
  signal?: AbortSignal,
): Promise<KnowledgeSummaryResponse> {
  return apiPost<KnowledgeSummaryResponse>('/knowledge-summary', payload, signal);
}

/* ============ AI 解析 ============ */

export interface ErrorParsePayload {
  question_text: string;
  student_answer?: string;
  correct_answer?: string;
  matched_knowledge?: {
    name: string;
    definition: string;
    error_tip: string;
  };
}

export interface ErrorParseResponse {
  /** markdown / 纯文本均可；前端直接渲染 */
  explanation: string;
  /** 关键步骤（可选） */
  steps?: string[];
  /** 复习建议（可选） */
  review_suggestion?: string;
  from_llm?: boolean;
}

/** POST /api/error-parse —— 单题 AI 解析 */
export function parseError(payload: ErrorParsePayload, signal?: AbortSignal): Promise<ErrorParseResponse> {
  return apiPost<ErrorParseResponse>('/error-parse', payload, signal);
}
